File: binding/include/libqopenglcontext_platform.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qnativeinterface-qwglcontext.html
pub const qnativeinterface__qwglcontext = struct {

    /// New constructs a new QNativeInterface::QWGLContext object.
    pub fn New() QtC.QNativeInterface__QWGLContext {
        return qtc.QNativeInterface__QWGLContext_new();
    }


    pub fn OpenGLModuleHandle() struct HINSTANCE__ {
        return @ptrCast(qtc.QNativeInterface__QWGLContext_OpenGLModuleHandle());
    }

    pub fn FromNative(context: *struct HGLRC__, window: *struct HWND__) QtC.QOpenGLContext {
        return qtc.QNativeInterface__QWGLContext_FromNative(@ptrCast(context), @ptrCast(window));
    }

    pub fn NativeContext(self: ?*anyopaque, ) struct HGLRC__ {
        return @ptrCast(qtc.QNativeInterface__QWGLContext_NativeContext(@ptrCast(self)));
    }

    pub fn OnNativeContext(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QNativeInterface__QWGLContext_Connect_NativeContext(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Allows for overriding the related default method
    pub fn OnNativeContext(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) struct HGLRC__) void {
        qtc.QNativeInterface__QWGLContext_OnNativeContext(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseNativeContext(self: ?*anyopaque, ) struct HGLRC__ {
        return @ptrCast(qtc.QNativeInterface__QWGLContext_QBaseNativeContext(@ptrCast(self)));
    }

    pub fn FromNative3(context: *struct HGLRC__, window: *struct HWND__, shareContext: ?*anyopaque) QtC.QOpenGLContext {
        return qtc.QNativeInterface__QWGLContext_FromNative3(@ptrCast(context), @ptrCast(window), @ptrCast(shareContext));
    }
};

File: binding/src/libqopenglcontext_platform.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_QNativeInterface__QWGLContext
#include <QOpenGLContext>
#include <qopenglcontext_platform.h>
#include "libqopenglcontext_platform.h"
#include "libqopenglcontext_platform.hxx"

QNativeInterface__QWGLContext* QNativeInterface__QWGLContext_new() {
	 return new VirtualQNativeInterfaceQWGLContext();
}

struct HINSTANCE__* QNativeInterface__QWGLContext_OpenGLModuleHandle() {
	return QNativeInterface::QWGLContext::openGLModuleHandle();
}

QOpenGLContext* QNativeInterface__QWGLContext_FromNative(struct HGLRC__* context, struct HWND__* window) {
	return QNativeInterface::QWGLContext::fromNative(context, window);
}

struct HGLRC__* QNativeInterface__QWGLContext_NativeContext(const QNativeInterface__QWGLContext* self) {
	return self->nativeContext();
}

void QNativeInterface__QWGLContext_Connect_NativeContext(QNativeInterface__QWGLContext* self, intptr_t slot) {
    void (*slotFunc)(QNativeInterface__QWGLContext*) = reinterpret_cast<void (*)(QNativeInterface__QWGLContext*)>(slot);
    QNativeInterface::QWGLContext::connect(self, &QNativeInterface::QWGLContext::nativeContext, [self, slotFunc]() {
	slotFunc(self);
    });
}

QOpenGLContext* QNativeInterface__QWGLContext_FromNative3(struct HGLRC__* context, struct HWND__* window, QOpenGLContext* shareContext) {
	return QNativeInterface::QWGLContext::fromNative(context, window, shareContext);
}

// Base class handler implementation
struct HGLRC__* QNativeInterface__QWGLContext_QBaseNativeContext(const QNativeInterface__QWGLContext* self) {
	auto* vqnativeinterface__qwglcontext = dynamic_cast<const VirtualQNativeInterfaceQWGLContext*>(self);
	if (vqnativeinterface__qwglcontext && vqnativeinterface__qwglcontext->isVirtualQNativeInterfaceQWGLContext) {
vqnativeinterface__qwglcontext->setQNativeInterface__QWGLContext_NativeContext_IsBase(true);
	return vqnativeinterface__qwglcontext->nativeContext();
}
}

// Auxiliary method to allow providing re-implementation
void QNativeInterface__QWGLContext_OnNativeContext(const QNativeInterface__QWGLContext* self, intptr_t slot) {
	auto* vqnativeinterface__qwglcontext = dynamic_cast<const VirtualQNativeInterfaceQWGLContext*>(self);
	if (vqnativeinterface__qwglcontext && vqnativeinterface__qwglcontext->isVirtualQNativeInterfaceQWGLContext) {
vqnativeinterface__qwglcontext->setQNativeInterface__QWGLContext_NativeContext_Callback(reinterpret_cast<VirtualQNativeInterfaceQWGLContext::QNativeInterface__QWGLContext_NativeContext_Callback>(slot));
}
}

